packages/core/src/components/ui/Link/Link.tsx

```diff
@@ -20,8 +20,10 @@ const Link = forwardRef<HTMLAnchorElement, LinkProps>(function Link<
     [href]
   )
 
+  // @TODO fix ts-ignore errors
   if (isInternalLink) {
     return (
+      // @ts-ignore
       <UILink
```

Comment:
Maybe removing the forwardRef for React 19 helps in this error. Polymorphic components get significantly simpler without the forwardRef function.